### PR TITLE
Code cleanup, small fixes (part 2)

### PR DESCRIPTION
**Pull request by gx740**
Note the IPcapDevice::getStatistics pure virtual method has been changed. A qualifier const was added.
It may lead to compilation errors if the user class is derived from this base class. But I think it should be done.

### CODE REVIEW

**Comment by seladb** [comment]
Are these changes still relevant after the previous code cleanup PR? Can you please verify?

**Comment by gx740** [comment]
Yes, the changes are relevant. But PR #283 is more important.

**Comment by seladb** [comment]
So the changes here don't overlap the previous PR?

**Comment by gx740** [comment]
I have successfully pulled the changes from seladb/dev. Initially, these two PR did't cross.

File: Pcap++/src/PfRingDevice.cpp [hunk 1]
```diff
@@ -274,9 +273,9 @@ bool PfRingDevice::openMultiRxChannels(uint8_t numOfRxChannelsToOpen, ChannelDis
 
 	if (ringsOpen < numOfRxChannelsToOpen)
 	{
-	    for (uint8_t i = 0; i < ringsOpen; i++)
-	    	pfring_close(m_PfRingDescriptors[i]);
-	    return false;
+	  for (uint8_t i = 0; i < ringsOpen; i++)
+	    pfring_close(m_PfRingDescriptors[i]);
+	  return false;
```
Comment (seladb) [inline Pcap++/src/PfRingDevice.cpp]: Please fix indentation here

Comment (gx740) [reply Pcap++/src/PfRingDevice.cpp]: Done

File: Pcap++/header/PcapLiveDevice.h [hunk 2]
```diff
@@ -222,60 +222,60 @@ namespace pcpp
...
 		/**
 		 * @return The device's link layer type
 		 */
-		virtual inline LinkLayerType getLinkType() { return m_LinkType; }
+		virtual LinkLayerType getLinkType() const { return m_LinkType; }
 		/**
 		 * @return A vector containing all addresses defined for this interface, each in pcap_addr_t struct
 		 */
-		inline std::vector<pcap_addr_t>& getAddresses() { return m_Addresses; }
+		std::vector<pcap_addr_t>& getAddresses() { return m_Addresses; } // TODO: Should getAddress really return non-const reference?
```
Comment (gx740) [inline Pcap++/header/PcapLiveDevice.h]: Pay attention on comment

File: Pcap++/header/PcapLiveDeviceList.h [hunk 3]
```diff
@@ -54,48 +54,48 @@ namespace pcpp
...
 		 */
-		PcapLiveDevice* getPcapLiveDeviceByName(const std::string& name);
+		PcapLiveDevice* getPcapLiveDeviceByName(const std::string& name) const;
 
 		/**
 		 * @return A list of all DNS servers defined for this machine. If this list is empty it means no DNS servers were defined or they
 		 * couldn't be extracted from some reason
 		 */
-		std::vector<IPv4Address>& getDnsServers();
+		std::vector<IPv4Address>& getDnsServers() { return m_DnsServers; } // TODO: Should this method really return a non-const reference?
```
Comment (gx740) [inline Pcap++/header/PcapLiveDeviceList.h]: Pay attention on comment

File: Pcap++/header/PfRingDeviceList.h [hunk 4]
```diff
@@ -46,7 +46,7 @@ namespace pcpp
 		 * Return a list of all available PF_RING devices
 		 * @return a list of all available PF_RING devices
 		 */
-		inline const std::vector<PfRingDevice*>& getPfRingDevicesList() { return m_PfRingDeviceList; }
+		const std::vector<PfRingDevice*>& getPfRingDevicesList() { return m_PfRingDeviceList; } // TODO: Should this method really return a non-const reference?
```
Comment (gx740) [inline Pcap++/header/PfRingDeviceList.h]: Pay attention on comment

File: Pcap++/header/PcapLiveDevice.h (hunk 2 above)
Comment (seladb) [reply Pcap++/header/PcapLiveDevice.h]: This list is read-only and shouldn't be changed by the caller so I guess the answer to your question is "no" - we should probably make it a const reference

File: Pcap++/header/PcapLiveDeviceList.h (hunk 3 above)
Comment (seladb) [reply Pcap++/header/PcapLiveDeviceList.h]: This list is read-only and shouldn't be changed by the caller so I guess the answer to your question is "no" - we should probably make it a const reference

File: Pcap++/header/PfRingDeviceList.h (hunk 4 above)
Comment (seladb) [reply Pcap++/header/PfRingDeviceList.h]: This list is read-only and shouldn't be changed by the caller so I guess the answer to your question is "no" - we should probably make it a const reference

File: Pcap++/header/PcapLiveDevice.h (hunk 2 above)
Comment (gx740) [reply Pcap++/header/PcapLiveDevice.h]: The changes have been made

**Pull request merged**